consensus/src/consensus/cache_policy_builder.rs

```diff
@@ -0,0 +1,144 @@
+use kaspa_database::prelude::CachePolicy;
+use kaspa_utils::mem_size::MemMode;
+use rand::Rng;
+
+/// Adds stochastic noise to cache sizes to avoid predictable and equal sizes across all network nodes
+fn noise(size: usize, magnitude: usize) -> usize {
+    if size == 0 {
+        // no noise if original size is zero
+        size
+    } else {
+        size + rand::thread_rng().gen_range(0..16) * magnitude
+    }
+}
+
+/// Bounds the size according to the "memory budget" (represented in bytes) and the approximate size of each unit in bytes
+fn bounded_size(desired_units: usize, memory_budget_bytes: usize, approx_unit_bytes: usize) -> usize {
+    let max_size = memory_budget_bytes / approx_unit_bytes;
+    usize::min(desired_units, max_size)
+}
+
+pub struct CachePolicyBuilder {
+    bytes_budget: usize,
+    max_items: usize,
+    min_items: usize,
+    unit_bytes: Option<usize>,
+    tracked: bool,
+    mem_mode: MemMode,
+}
+
+impl Default for CachePolicyBuilder {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl CachePolicyBuilder {
+    pub fn new() -> Self {
+        Self {
+            bytes_budget: usize::MAX,
```

Comment:
nit: It seems the `MAX` value is never passed to the resulted `CachePolicy`, so I would use `Option<usize>` instead